src/app/lib/utilities/preprocessor/rules/cpstypes.js

```diff
@@ -0,0 +1,30 @@
+import deepGet from '../../../../helpers/json/deepGet';
+
+const whitelist = ['STY', 'MAP'];
+
+const filterUnknownCpsTypes = data => {
+  const groups = deepGet(['content', 'groups'], data);
+
+  if (!groups) {
+    return data;
+  }
+
+  const newGroups = groups.map(group => {
+    // avoid eslint complaints about 'no-param-reassign'
+    const newGroup = group;
+    const newItems =
+      group.items &&
```

Comment:
```suggestion
      Array.isArray(group.items) &&
```
Can you make a quick tests to check this logic please?

Reply:
Ah! good point! Didn't fix it quite as you suggested though; this would replace eg. `items: undefined` (or missing items field) with `items: false`. I went with

```js
if (Array.isArray(group.items)) {
  newGroup.items = group.items.filter(
    item => item.cpsType && whitelist.includes(item.cpsType),
  );
}
```

which should leave non-array items as they are (or not set them), allowing the group to be caught by #1770.